### PR TITLE
feat: implement `atand` intrinsic function

### PR DESCRIPTION
**Pull request by parth121101**

### CODE REVIEW

File: integration_tests/intrinsics_209.f90 [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                                        -0.994005994005994_dp]
+         
+    real(dp), parameter :: res(19) = atand(w_comp)
+
+    w = [0.321_dp, 0.526_dp, 0.728_dp, -0.321_dp, -0.526_dp, -0.728_dp, &
+         -1.0_dp, -0.9995004995004995_dp, -0.999000999000999_dp, &
+         -0.9985014985014985_dp, -0.998001998001998_dp, -0.9975024975024975_dp, &
+         -0.997002997002997_dp, -0.9965034965034965_dp, -0.996003996003996_dp, &
+         -0.9955044955044955_dp, -0.995004995004995_dp, -0.9945054945054945_dp, &
+         -0.994005994005994_dp]
```
Comment (Pranavchiku) [inline integration_tests/intrinsics_209.f90:23]: domain for `atand` is `(-inf, inf)`, let's add test for large negative, negative, positive, large positive numbers.

File: integration_tests/intrinsics_209.f90 [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    if (atand(z) - (-28.4576912) > 1e-5) error stop
+
+    print *, atand(-0.123_sp)
+    if (atand(-0.123_sp) - (-7.01216030) > 1e-5) error stop
+
+    print *, atand(-0.876_sp)
+    if (atand(-0.876_sp) - (-41.2183571) > 1e-5) error stop
+
+    print *, atand(-0.542_sp)
+    if (atand(-0.542_sp) - (-28.4576931) > 1e-5) error stop
```
Comment (Pranavchiku) [inline integration_tests/intrinsics_209.f90:85]: I think the array based testing is sufficient, no need to explicitly test like these.

File: integration_tests/intrinsics_209.f90 (hunk 1 above)
Comment (parth121101) [reply integration_tests/intrinsics_209.f90:23]: Sure @Pranavchiku , thanks for revising my functions concepts.

File: src/libasr/pass/intrinsic_functions.h [hunk 3]
```diff
@@ -3179,6 +3180,50 @@ namespace Acosd {
...
+            fn_symtab->add_symbol(c_func_name, s);
+            dep.push_back(al, s2c(al, c_func_name));
+            body.push_back(al, b.Assignment(result, b.Call(s, args, return_type)));
+        }
+
+        ASR::symbol_t *new_symbol = make_ASR_Function_t(fn_name, fn_symtab, dep, args,
+            body, result, ASR::abiType::Source, ASR::deftypeType::Implementation, nullptr);
+        scope->add_symbol(fn_name, new_symbol);
+        return b.Call(new_symbol, new_args, return_type);
+    }
```
Comment (Pranavchiku) [inline src/libasr/pass/intrinsic_functions.h:3223]: For functions using `math.h` to compute value, let's create a macro like `create_trig`. I think this should be done at high priority in subsequent PR.

Comment (Pranavchiku) [reply src/libasr/pass/intrinsic_functions.h:3223]: Created https://github.com/lfortran/lfortran/issues/3857 for this.

File: integration_tests/intrinsics_209.f90 [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    x = [0.321_sp, 0.526_sp, 0.728_sp, -0.321_sp, -0.526_sp, -0.728_sp, &
+         -1.0_sp, -0.999_sp, -0.999_sp, -0.998_sp, -0.998_sp, -0.997_sp, &
+         -0.997_sp, -0.996_sp, -0.996_sp, -0.995_sp, -0.995_sp, -0.994_sp, -0.994_sp]
+
+    large_pos = [988.522977022977_dp, 989.021978021978_dp, 989.520979020979_dp ,990.01998001998_dp , & 
+            990.518981018981_dp, 991.017982017982_dp, 991.516983016983_dp, 992.015984015984_dp, &
+            992.514985014985_dp, 993.013986013986_dp, 993.512987012987_dp, 994.011988011988_dp, &
+            994.510989010989_dp, 995.00999000999_dp, 995.508991008991_dp, 996.007992007992_dp, &
+            996.506993006993_dp, 997.005994005994_dp, 997.504995004995_dp]
```
Comment (Pranavchiku) [inline integration_tests/intrinsics_209.f90:38]: these shall be tested compile time as well

File: integration_tests/intrinsics_209.f90 [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    
+    large_neg = [-988.522977022977_dp, -989.021978021978_dp, -989.520979020979_dp, -990.01998001998_dp , & 
+            -990.518981018981_dp, -991.017982017982_dp, -991.516983016983_dp, -992.015984015984_dp, &
+            -992.514985014985_dp, -993.013986013986_dp, -993.512987012987_dp, -994.011988011988_dp, &
+            -994.510989010989_dp, -995.00999000999_dp, -995.508991008991_dp, -996.007992007992_dp, &
+            -996.506993006993_dp, -997.005994005994_dp, -997.504995004995_dp]
```
Comment (Pranavchiku) [inline integration_tests/intrinsics_209.f90:44]: just merge all these in a single array, test for both compile and runtime.

**Review by Pranavchiku — DISMISSED** [review]
Left comments above

**Comment by certik** [comment]
Conflicts have to be resolved.

File: integration_tests/CMakeLists.txt [hunk 6]
```diff
@@ -712,33 +712,34 @@ RUN(NAME intrinsics_178 LABELS gfortran llvm llvm_wasm) # rrspacing
...
-RUN(NAME intrinsics_201 LABELS gfortran llvm llvm_wasm) # any
-RUN(NAME intrinsics_202 LABELS gfortran llvm llvm_wasm)
-RUN(NAME intrinsics_203 LABELS gfortran llvm llvm_wasm) # asind
-RUN(NAME intrinsics_204 LABELS gfortran llvm llvm_wasm) # bessely1
-RUN(NAME intrinsics_205 LABELS gfortran llvm llvm_wasm) # acosd
-RUN(NAME intrinsics_206 LABELS gfortran llvm llvm_wasm fortran) # poppar
-RUN(NAME intrinsics_207 LABELS gfortran llvm llvm_wasm) # double precision intrinsics
-RUN(NAME intrinsics_208 LABELS gfortran llvm llvm_wasm) # bit_size
+RUN(NAME intrinsics_182 LABELS gfortran llvm) # dshiftl
+RUN(NAME intrinsics_183 LABELS gfortran llvm) # ibits
```
Comment (Pranavchiku) [inline integration_tests/CMakeLists.txt:716]: any reason for removing `llvm_wasm`?

**Review by certik — APPROVED** [review]
